e2e/customized-env/customized_env_test.go

```diff
@@ -0,0 +1,265 @@
+		})
+
+		It("svc init should create a svc manifest", func() {
+			Expect("./copilot/front-end/manifest.yml").Should(BeAnExistingFile())
+		})
+
+		It("svc ls should list the svc", func() {
+			svcList, svcListError := cli.SvcList(appName)
+			Expect(svcListError).NotTo(HaveOccurred())
+			Expect(len(svcList.Services)).To(Equal(1))
+			Expect(svcList.Services[0].Name).To(Equal("front-end"))
+		})
+	})
+
+	Context("when deploying a svc to test and prod envs", func() {
+		var (
+			testDeployErr    error
+			prodEndDeployErr error
+			svcName          string
+		)
+		BeforeAll(func() {
+			svcName = "front-end"
+			_, testDeployErr = cli.SvcDeploy(&client.SvcDeployInput{
+				Name:     svcName,
+				EnvName:  "test",
+				ImageTag: "gallopinggurdey",
+			})
+
+			_, prodEndDeployErr = cli.SvcDeploy(&client.SvcDeployInput{
+				Name:     svcName,
+				EnvName:  "prod",
+				ImageTag: "gallopinggurdey",
+			})
+		})
+
+		It("svc deploy should succeed to both environment", func() {
+			Expect(testDeployErr).NotTo(HaveOccurred())
+			Expect(prodEndDeployErr).NotTo(HaveOccurred())
+		})
+
+		It("svc show should include a valid URL and description for test and prod envs", func() {
+			svc, svcShowErr := cli.SvcShow(&client.SvcShowRequest{
+				AppName: appName,
+				Name:    svcName,
+			})
+			Expect(svcShowErr).NotTo(HaveOccurred())
+			Expect(len(svc.Routes)).To(Equal(2))
+			// Group routes by environment
+			envRoutes := map[string]client.SvcShowRoutes{}
+			for _, route := range svc.Routes {
+				envRoutes[route.Environment] = route
+			}
+
+			Expect(len(svc.ServiceDiscoveries)).To(Equal(1))
+			Expect(svc.ServiceDiscoveries[0].Environment).To(ConsistOf("test", "prod"))
+			Expect(svc.ServiceDiscoveries[0].Namespace).To(Equal(fmt.Sprintf("%s.%s.local:80", svc.SvcName, appName)))
+
+			// Call each environment's endpoint and ensure it returns a 200
+			for _, env := range []string{"test", "prod"} {
+				route := envRoutes[env]
+				Expect(route.Environment).To(Equal(env))
+				Eventually(func() (int, error) {
+					resp, fetchErr := http.Get(route.URL)
+					return resp.StatusCode, fetchErr
+				}, "30s", "1s").Should(Equal(200))
+			}
+		})
+
+		It("svc logs should display logs", func() {
+			for _, envName := range []string{"test", "prod"} {
+				var svcLogs []client.SvcLogsOutput
+				var svcLogsErr error
+				Eventually(func() ([]client.SvcLogsOutput, error) {
+					svcLogs, svcLogsErr = cli.SvcLogs(&client.SvcLogsRequest{
+						AppName: appName,
+						Name:    svcName,
+						EnvName: envName,
+						Since:   "1h",
+					})
+					return svcLogs, svcLogsErr
+				}, "60s", "10s").ShouldNot(BeEmpty())
+
+				for _, logLine := range svcLogs {
+					Expect(logLine.Message).NotTo(Equal(""))
+					Expect(logLine.LogStreamName).NotTo(Equal(""))
+					Expect(logLine.Timestamp).NotTo(Equal(0))
+					Expect(logLine.IngestionTime).NotTo(Equal(0))
+				}
+			}
+		})
+
+		It("env show should display info for test and prod envs", func() {
```

Comment:
Dumb question: Do we we care about any of the service workflows in this e2e test?

I wonder if instead we should have only this `It` and include `--resources` to the command to validate if the environment is created with the appropriate resource configurations?

Reply:
hmm i think it doesn't hurt to validate show and logs to make sure the svc we deploy is actually working as expected. But I'll add validation for `--resources`!